R/tidy_upload.R

```diff
@@ -15,133 +45,125 @@ clean_dataframe <- function(data){
-      dplyr::mutate(fft = dplyr::case_when(
-        fft %in% c("Dont know", "Dont Know") ~ NA_integer_,
-        fft == "Very poor" ~ 1L,
-        fft == "Poor" ~ 2L,
-        fft == "Neither good nor poor" ~ 3L, 
-        fft == "Good" ~ 4L,
-        fft == "Very good" ~ 5L,
-        TRUE ~ NA_integer_
-      )) %>%
-      dplyr::mutate(age = dplyr::case_when(
-        age == "Up to 25" ~ "0 - 25",
-        TRUE ~ age)) %>% 
-      dplyr::mutate(date = as.Date(date, format = "%d/%m/%Y"))
-  }
+  # do trust specific data cleaning ----
   
-  if(trust_id == "trust_d"){
-    
-    db_tidy <- db_tidy %>% 
-      dplyr::mutate(fft = 6 - fft) %>%
-      dplyr::mutate(date = lubridate::as_date(date, format = "%m/%d/%Y"))
-  }
+  if (trust_id == "trust_c") db_tidy <- db_tidy %>% tidy_trust_c_e()
+  if (trust_id == "trust_d")  db_tidy <- db_tidy %>% tidy_trust_d()
+  if (trust_id == "trust_e") db_tidy <- db_tidy %>% tidy_trust_c_e()
+  
+  # call API for label predictions ----
+  
+  # convert data to json for API as specified in the API doc
+  json_data <- db_tidy  |>
+    dplyr::select(comment_id, comment_text) |>
+    jsonlite::toJSON()
+  
+  preds <-  api_pred("http://127.0.0.1:8000/predict_multilabel", json=json_data) %>% 
+    dplyr::mutate(comment_id = as.integer(comment_id))
+  
+  print('Done with API call ...')
+  
+  # rename the columns to make the data compatible with old data format currently in use
+  
+  final_df <- db_tidy %>% dplyr::left_join(preds, by = c('comment_id', 'comment_text')) %>% 
+    dplyr::rename(fft = fft_score, category = labels,
+                  comment_txt = comment_text, 
+    ) %>% 
+    dplyr::select(-comment_id)  %>% 
+    dplyr::mutate(hidden = 0,
+                  date = as.Date(.data$date)
+    ) %>%
+    tidy_label_column('category') 
+  
+  # get the current maximum  row_id value in the database table
+  
+  max_id <- DBI::dbGetQuery(conn, paste0("SELECT MAX(row_id) FROM ", trust_id))$`MAX(row_id)`
+  
+  # set the starting value for the auto-incremented row_id
+  # This will ensure that when we append the new data, 
+  # the row_id values will be sequential and there will be no gaps.
+  final_df <- final_df %>% 
+    dplyr::mutate(
+      row_id = seq.int(max_id + 1, max_id + nrow(.)),
+      comment_type = stringr::str_replace_all(.data$comment_type, 'question', 'comment')
+    )
+  
+  # write the processed data to database
+  print('Just started appending to database ...')
+  
+  # DBI::dbWriteTable(conn, trust_id,  final_df, append = TRUE) # this doesn't throw error when data can't be read e.g dues to mismatch datatypes.
+  
+  # this throw error when data can't be appended e.g when data column can't be coerce into the db table datatype
+  dplyr::rows_append(
+    dplyr::tbl(conn, trust_id),
+    final_df,
+    copy = TRUE,
+    in_place = TRUE
+  )
+  
+  print('Done appending to database ...')
+  
+  # reset the db data ------------------- remove later ------------------------
```

Comment:
There is functionality in golem where you have different configuration for dev versus prod, so you could think about using that here

Reply:
I can't figure this out for now. I will deal with it  later